### PR TITLE
Toggle Updates - Angular

### LINKED ISSUE #3296: Toggle Upgrades - Part 2 (Angular)
**Notes:**
React will be completed in #3295 

**AC:**
# Angular
- Add ability to have the toggle content open by default - use `isOpen` to match React API
- add type=button
- change `title` prop to be `triggerText` since title causes a11y issues.
- add event emitter for when toggle opens or closes ex.
```
  /**
   * When the toggle is opened this event will be emitted.
   */
  @Output() openedEvent = new EventEmitter<any>();

  /**
   * When the toggle is closed this event will be emitted.
   */
  @Output() closedEvent = new EventEmitter<any>();
```
- add ability to enter a different icon name
- add ability to add different classes to the icon
- add ability to add classes to the content

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
- [x] Add ability to have the toggle content open by default - match React naming `isOpen`.
- [x] add type=button
- [x] Add `triggerText` input - continue support for `title` until next release.
- [x] add event emitter for when toggle opens or closes
- [x] add `toggleIconName` input
- [x] add `iconAdditionalClasses` input - continue support for `iconClass` until next release.
- [x] add `titleAdditionalClasses` input - continue support for `titleFontClass` until next release.
- [x] add `contentAdditionalClasses` input
- [x] New follow up issue to remove deprecated inputs - `iconClass`, `title`, `titleFontClass`
https://github.com/sparkdesignsystem/Backlog/issues/1305

### Associated Issue
Fixes #3296 

### Documentation
 - [x] Update Spark Docs

### Code
 - [x] Build Component in Angular
 - [x] Unit Testing in Angular with `npm run test` in `angular/` (100% coverage, 100% passing)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Google Chrome (Mobile)
  - [x] Mozilla Firefox
  - [x] Mozilla Firefox (Mobile)
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari
  - [ ] Apple Safari (Mobile)
  
### Accessibility Testing
  - [ ] Axe browser extension
  - [ ] Jaws Inspect
  - [ ] VoiceOver (iOS) or Talkback (Android)

### Deploy Preview Reviewed and Approved
 - [ ] Design Team
 - [ ] Accessibility Team

### Screenshots
image

### CODE REVIEW

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 1]
```diff
@@ -56,26 +58,69 @@ export class SprkToggleComponent implements AfterViewInit {
    */
   @Input()
   additionalClasses: string;
+  // TODO: Remove `title` in issue #1305
   /**
+   * Deprecated - Use `triggerText` instead.
    * The value supplied will be
    * rendered as the main Toggle link text.
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts:65]:
```suggestion
   * rendered as the main Toggle button text.
```

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 2]
```diff
@@ -56,26 +58,69 @@ export class SprkToggleComponent implements AfterViewInit {
...
    */
   @Input()
   title: string;
   /**
-   * The value supplied will be assigned as a
-   * CSS class on the icon used in the Toggle.
-   * This is intended for overrides.
+   * The value supplied will be
+   * rendered as the main Toggle link text.
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts:71]:
```suggestion
   * rendered as the main Toggle button text.
```

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 3]
```diff
@@ -110,13 +166,13 @@ export class SprkToggleComponent implements AfterViewInit {
...
-    this.isOpen === false
-      ? (this.iconStateClass = '')
-      : (this.iconStateClass = 'sprk-c-Icon--open');
+    if (this.isOpen) {
+      this.animState = 'open';
+      this.iconStateClass = 'sprk-c-Icon--open';
+    } else {
+      this.animState = 'closed';
+      this.iconStateClass = '';
+    }
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts:175]: I like this much better than the previous version!

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts [hunk 4]
```diff
@@ -44,37 +40,80 @@ describe('SprkToggleComponent', () => {
...
     element.querySelector('button').click();
     element.querySelector('button').click();
     fixture.detectChanges();
     expect(
-      element.querySelector('button .sprk-c-Icon').classList.toString()
+      element.querySelector('button .sprk-c-Icon').classList.toString(),
     ).toEqual('sprk-c-Icon sprk-c-Icon--xl sprk-u-mrs sprk-c-Icon--toggle');
   });
 
+  it('should render open if isOpen is set to true', () => {
```
Comment (RVMendoza) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts:69]: This is all semantics, but I wonder if this test is really "should load correct classes if isOpen is set to true"

**Review by bobwise — APPROVED** [review]

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts [hunk 5]
```diff
@@ -56,26 +58,69 @@ export class SprkToggleComponent implements AfterViewInit {
...
   titleFontClass = 'sprk-b-TypeBodyThree';
+  // TODO: Move the default value from titleFontClass to
+  // titleAdditionalClasses in future issue #1305.
+  /**
+   * Expects a space separated string
+   * of classes to be added to the
+   * trigger text.
+   */
+  @Input()
+  titleAdditionalClasses: string;
```
Comment (afebbraro) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts:123]: hey @guern1kn , In going through this, a light bulb went off haha. If we change title to heading we have to change the other names too like titleAdditionalClasses

**Review by afebbraro — CHANGES_REQUESTED** [review]

File: angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts [hunk 6]
```diff
@@ -44,37 +40,84 @@ describe('SprkToggleComponent', () => {
...
+    fixture.detectChanges();
+    expect(
+      element.querySelector('button .sprk-c-Icon').classList.toString(),
+    ).toEqual(
+      'sprk-c-Icon sprk-c-Icon--xl sprk-u-mrs sprk-c-Icon--toggle sprk-c-Icon--open',
+    );
+    expect(contentElement.getAttribute('style')).toContain(
+      'visibility:visible',
+    );
+    expect(triggerElement.getAttribute('aria-expanded')).toEqual('true');
```
Comment (bobwise) [inline angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.spec.ts:80]: 🎉

**Review by bobwise — APPROVED** [review]

**Review by RVMendoza — APPROVED** [review]

**Review by afebbraro — APPROVED** [review]

**Pull request merged**